lisa/util/shell.py

```diff
@@ -43,14 +46,136 @@ def close(self) -> None:
 
     def mkdir(
         self,
-        path: Path,
+        path: PurePath,
         mode: int = 0o777,
         parents: bool = True,
         exist_ok: bool = False,
     ) -> None:
         self.initialize()
         if self.isRemote:
             assert self.innerShell
-            self.innerShell.mkdir(path, mode=mode, parents=parents, exist_ok=exist_ok)
+            path_str = self._purePathToStr(path)
+            self.innerShell.mkdir(
+                path_str, mode=mode, parents=parents, exist_ok=exist_ok
+            )
         else:
+            assert isinstance(path, Path)
             path.mkdir(mode=mode, parents=parents, exist_ok=exist_ok)
+
+    def exists(self, path: PurePath) -> bool:
+        self.initialize()
+        exists = False
+        if self.isRemote:
+            assert self.innerShell
+            path_str = self._purePathToStr(path)
+            exists = self.innerShell.exists(path_str)
+        else:
+            assert isinstance(path, Path)
+            exists = path.exists()
+        return exists
+
+    def remove(self, path: PurePath, recursive: bool = False) -> None:
+        self.initialize()
+        if self.isRemote:
+            assert self.innerShell
+            path_str = self._purePathToStr(path)
+            self.innerShell.remove(path_str, recursive)
+        else:
+            assert isinstance(path, Path)
+            path.rmdir()
+
+    def chmod(self, path: PurePath, mode: int) -> None:
+        self.initialize()
+        if self.isRemote:
+            assert self.innerShell
+            path_str = self._purePathToStr(path)
```

Comment:
Just cause I'm noticing a mix, what did we choose for naming conventions? PEP8 generally says to use snake_case for variables and functions https://www.python.org/dev/peps/pep-0008/#function-and-variable-names

Reply:
Thank you. I will change all method and variable names to use under_score instead of capWords.

Reply:
I'm avoiding to add too many underscores in names, feel free to add some, if it's hard to know.

Reply:
BTW, what tool can prevent it happening?